diff --git a/src/server/scripts/Northrend/dalaran.cpp b/src/server/scripts/Northrend/dalaran.cpp
index c01d348..45db116 100644
--- a/src/server/scripts/Northrend/dalaran.cpp
+++ b/src/server/scripts/Northrend/dalaran.cpp
@@ -77,7 +77,7 @@ public:
             switch (me->GetEntry())
             {
                 case 29254:
-                    if (pPlayer->GetTeam() == HORDE)              // Horde unit found in Alliance area
+                    if (pPlayer->GetTeam() == HORDE && !pPlayer->HasAura(70971) && !pPlayer->HasAura(70972))              // Horde unit found in Alliance area
                     {
                         if (GetClosestCreatureWithEntry(me, NPC_APPLEBOUGH_A, 32.0f))
                         {
@@ -89,7 +89,7 @@ public:
                     }
                     break;
                 case 29255:
-                    if (pPlayer->GetTeam() == ALLIANCE)           // Alliance unit found in Horde area
+                    if (pPlayer->GetTeam() == ALLIANCE && !pPlayer->HasAura(70973) && !pPlayer->HasAura(70974))           // Alliance unit found in Horde area
                     {
                         if (GetClosestCreatureWithEntry(me, NPC_SWEETBERRY_H, 32.0f))
                         {
