diff --git a/src/server/game/Server/Protocol/Handlers/QuestHandler.cpp b/src/server/game/Server/Protocol/Handlers/QuestHandler.cpp
index 24cfea3..b93c24b 100755
--- a/src/server/game/Server/Protocol/Handlers/QuestHandler.cpp
+++ b/src/server/game/Server/Protocol/Handlers/QuestHandler.cpp
@@ -119,8 +119,8 @@ void WorldSession::HandleQuestgiverAcceptQuestOpcode(WorldPacket & recv_data)
     uint32 unk1;
     recv_data >> guid >> quest >> unk1;
 
-    if (!GetPlayer()->isAlive())
-        return;
+    //if (!GetPlayer()->isAlive())
+        //return;
 
     sLog->outDebug(LOG_FILTER_NETWORKIO, "WORLD: Received CMSG_QUESTGIVER_ACCEPT_QUEST npc = %u, quest = %u, unk1 = %u", uint32(GUID_LOPART(guid)), quest, unk1);
 
@@ -135,6 +135,9 @@ void WorldSession::HandleQuestgiverAcceptQuestOpcode(WorldPacket & recv_data)
         return;
     }
 
+	if (!GetPlayer()->isAlive() && (!pObject->ToCreature() || !pObject->ToCreature()->GetCreatureInfo()->type_flags & CREATURE_TYPEFLAGS_GHOST))
+		return;
+
     Quest const* qInfo = sObjectMgr->GetQuestTemplate(quest);
     if (qInfo)
     {
@@ -294,8 +297,8 @@ void WorldSession::HandleQuestgiverChooseRewardOpcode(WorldPacket & recv_data)
         return;
     }
 
-    if (!GetPlayer()->isAlive())
-        return;
+    //if (!GetPlayer()->isAlive())
+        //return;
 
     sLog->outDebug(LOG_FILTER_NETWORKIO, "WORLD: Received CMSG_QUESTGIVER_CHOOSE_REWARD npc = %u, quest = %u, reward = %u", uint32(GUID_LOPART(guid)), quest, reward);
 
@@ -303,6 +306,9 @@ void WorldSession::HandleQuestgiverChooseRewardOpcode(WorldPacket & recv_data)
     if (!pObject)
         return;
 
+	if (!GetPlayer()->isAlive() && (!pObject->ToCreature() || !pObject->ToCreature()->GetCreatureInfo()->type_flags & CREATURE_TYPEFLAGS_GHOST))
+		return;
+
     if (!pObject->hasInvolvedQuest(quest))
         return;
 
@@ -349,8 +355,8 @@ void WorldSession::HandleQuestgiverRequestRewardOpcode(WorldPacket & recv_data)
     uint64 guid;
     recv_data >> guid >> quest;
 
-    if (!GetPlayer()->isAlive())
-        return;
+    //if (!GetPlayer()->isAlive())
+        //return;
 
     sLog->outDebug(LOG_FILTER_NETWORKIO, "WORLD: Received CMSG_QUESTGIVER_REQUEST_REWARD npc = %u, quest = %u", uint32(GUID_LOPART(guid)), quest);
 
@@ -358,6 +364,9 @@ void WorldSession::HandleQuestgiverRequestRewardOpcode(WorldPacket & recv_data)
     if (!pObject||!pObject->hasInvolvedQuest(quest))
         return;
 
+	if (!GetPlayer()->isAlive() && (!pObject->ToCreature() || !pObject->ToCreature()->GetCreatureInfo()->type_flags & CREATURE_TYPEFLAGS_GHOST))
+		return;
+
     if (_player->CanCompleteQuest(quest))
         _player->CompleteQuest(quest);
 
@@ -462,9 +471,17 @@ void WorldSession::HandleQuestgiverCompleteQuest(WorldPacket& recv_data)
     uint64 guid;
     recv_data >> guid >> quest;
 
-    if (!_player->isAlive())
+    //if (!_player->isAlive())
+        //return;
+
+    Object* pObject = ObjectAccessor::GetObjectByTypeMask(*_player, guid, TYPEMASK_UNIT|TYPEMASK_GAMEOBJECT);
+    if (!pObject)
         return;
 
+    if (!GetPlayer()->isAlive() && (!pObject->ToCreature() || !pObject->ToCreature()->GetCreatureInfo()->type_flags & CREATURE_TYPEFLAGS_GHOST))
+		return;
+
+
     sLog->outDebug(LOG_FILTER_NETWORKIO, "WORLD: Received CMSG_QUESTGIVER_COMPLETE_QUEST npc = %u, quest = %u", uint32(GUID_LOPART(guid)), quest);
 
     Quest const *pQuest = sObjectMgr->GetQuestTemplate(quest);
